### ISSUE TITLE
 Issues with recent changes to Ubuntu-Latest Github Hosted Runners

### ISSUE THREAD

**Issue by timharris777** [actions/virtual-environments#2234]
Recently the ubuntu-latest github hosted runners changed. The user and group for the workspace changed from `actions:actions` to `runner:docker` and it looks like `env` changed from only having actions related stuff in it to being polluted with all kinds of env vars from the host machine. These unannounced changes are breaking some of our workflows.

I have an enterprise support ticket opened where they indicated I should open an issue against this repo: https://enterprise.githubsupport.com/hc/en-us/requests/937432

Thanks!

**Comment by maxim-lobanov** [comment]
@timharris777 , thank you for logging issue.
No changes on image-generation side that could cause it since workspace directory doesn't exist during image generation and it is created later.
We will try to find and escalate it to appropriate  team.

**Comment by hross** [comment]
The last runner change was November 9th, so I feel pretty confident to say it doesn't have to do with runner changes. @maxim-lobanov can you set up a test somewhere that can repro the env issuess?

**Comment by maxim-lobanov** [comment]
I can confirm that currently permissions / owners are set `runner:docker`. It can retrieved with simple pipeline:
```
on: [push, workflow_dispatch]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - run: ls -la
```
![image](https://user-images.githubusercontent.com/16715858/101902184-4b204680-3bc3-11eb-827b-c54eecfa023b.png)

I have never checked it so can't confirm that it was `actions:actions` before and what could be changed.
GitHub workspace folder is created after image-generation (we don't have such folder on clean image).
I guess it is created when GitHub configures machine before build.

**Comment by maxim-lobanov** [comment]
@timharris777 , sorry, I don't have an access to support ticket for now so could you please share any additional information?
1) What use case is affected by this change and how did you notice it?
2) Do you use any third-party actions in your build workflow? (If it is possible, could you please share YAML file, you can cut sensitive information).
3) Could you please find the latest successful build and the first failed build (where this issue is reproduced) and share the log of initialize job step for both builds? It could help to determine whether the issue was started with image update or not
Example: 
![image](https://user-images.githubusercontent.com/16715858/101926398-28eaf080-3be4-11eb-8589-2e4ff3d538de.png)

**Comment by timharris777** [comment]
I have a custom action we created for running commands in docker images (https://github.com/chick-fil-a/gha-docker-run). The final step in the actions was to `chmod -R actions:actions .` because any actions taken in the docker image typically end up `root:root` which then cause github actions to blow up because it can't work with the files. Everything was working fine until recently when our workflows started blowing up. I looked into what was going on and the user:group used by actions had changed from `actions:actions` to `runner:docker`. I added a conditional statement in my action to account for this, but I thought it was strange (https://github.com/chick-fil-a/gha-docker-run/blob/badefa93b50689e2746f7fae68f5b781c8009f74/index.js#L65).

We noticed the issue and created the workaround on Monday 12/7/2020. I assumed something had changed over the weekend because the week prior everything was working fine for us. I went to grab the logs from the failed job that alerted us, but it looks like on of my guys "re-ran" the workflow after I updated the action and it wiped out the failed log.

**Comment by maxim-lobanov** [comment]
@timharris777 , thank you for providing links. it is a bit clearer now.

Based on your action,  the following code was executed for you previously, right?
```
core.startGroup('fixing permissions');
if (os.userInfo().username == "actions") {
    await exec.exec("sudo chown -R actions:actions .");
}
```

As I now, `os.userInfo().username` just prints the name of current user, analogue of $USERNAME env variable.
Based on it, I am pretty sure that we have never had `actions` username on Hosted agents in GitHub Actions and nothing was changed about it recently since a lot of logic depend on it.

As a proof, I have found old (1 month ago) build in actions/virtual-environments repo on ubuntu-latest: https://github.com/actions/virtual-environments/runs/1357109795?check_suite_focus=true
It has `runner` user name.
<img width="720" alt="Screen Shot 2020-12-14 at 3 29 55 PM" src="https://user-images.githubusercontent.com/16715858/102081568-81f39800-3e21-11eb-8cfe-a5a3f6624423.png">

May be your previous builds used self-hosted agent instead of Hosted one?
Can't say more without having logs of build with this issue.

**Comment by timharris777** [comment]
@maxim-lobanov , thanks for looking into it. Previously, I just had `await exec.exec("sudo chown -R actions:actions .")` without any conditions and it succeeded. Then it started failing saying the user and group didn't exist so I had to add the conditional statements.  That's very interesting... we specifically built our self-hosted runners to match how the github hosted runners were built and that is why we selected `actions:actions` as the user and group names. No matter though, the condition statements are preventing failures now so we are good to go. Thanks!
